pulsar-zookeeper-utils/src/main/java/com/yahoo/pulsar/zookeeper/ZooKeeperCache.java

```diff
@@ -166,6 +166,23 @@ private void invalidateExists(String path) {
         existsCache.invalidate(path);
     }
 
+    public void asyncInvalidate(String path) {
+        if (scheduledExecutor != null) {
```

Comment:
> Why the executors could be null?

because in case of [ZookeeperCacheLoader](https://github.com/yahoo/pulsar/blob/master/pulsar-discovery-service/src/main/java/com/yahoo/pulsar/discovery/service/web/ZookeeperCacheLoader.java#L69) we depend on ForkJoinPool.
Also, it can also possible if someone pass null executor in constructor: `ZooKeeperCache(ZooKeeper zkSession, OrderedSafeExecutor executor, ScheduledExecutorService scheduledExecutor) `